src/framework/preferences/js/PrimaryBuilder.js

```diff
@@ -77,22 +79,22 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
     /**
      * An invoker method that builds a list of grades that comprise a final version of the primary schema.
      * @param {JSON} schemaIndex - A global index of all schema grades registered with the framework.
-     * @param {Array} typeFilter   - A list of all necessarry top level preference names.
+     * @param {Array} preferences   - A list of all necessary top level preference names.
      * @param {JSON} primarySchema - Primary schema provided as an option to the primary builder.
      * @return {Array} - A list of schema grades.
      */
-    fluid.prefs.primaryBuilder.buildPrimary = function (schemaIndex, typeFilter, primarySchema) {
+    fluid.prefs.primaryBuilder.buildPrimary = function (schemaIndex, preferences, primarySchema) {
         var suppliedPrimaryGradeName = "fluid.prefs.schemas.suppliedPrimary" + fluid.allocateGuid();
```

Comment:
Whilst this is in progress, we should note that "supplied" is a quite misleading name for this since it is not supplied at all. Better to just use the name of this function itself as a clue to its origin, e.g. "buildPrimary"